--- a/keys.ino
+++ b/keys.ino
@@ -1,3 +1,6 @@
+void openPatternView();
+void openMenuView();
+void openSongView();
 
 void DO_KEYPAD(){
@@ -281,18 +284,9 @@
               break;
             // Song
             case 23:
-              if (modeZ==tMemo) {
-                modeZ=tMfirs;
-              } else if (modeZ==tMfirs) {
-                modeZ=tMlast;
-              } else if (modeZ==tMlast) {
-                modeZ=tMemo;
-              } else {
-                songing=!songing;
-                if (recording) songing=false; // if recording I cant set song mode
-              }
-              break;                      
-          }  
+              openSongView();
+              break;
+          }  
 
         // shift 
@@ -374,6 +368,12 @@
             }
             do_rot();
             break;
+          case 29:
+            openPatternView();
+            break;
+          case 30:
+            openMenuView();
+            break;
           case 31:
             shifting=!shifting;
             refresh_shift_key();
